html/components/autocomplete.stories.js

```diff
@@ -0,0 +1,240 @@
+
+#### Keyboard Interactions
+- The list can be closed by clicking on the document body or by pressing
+the \`Escape\` key.
+- When the list is open, items in the list can be highlighted using the
+\`Up\` and \`Down\` arrow keys. Note that this highlight is distinct from the
+browser focus, which stays in the input.
+
+#### Accessibility
+- The component should include an element with \`aria-live="polite"\`. The
+  contents of this element should be updated to reflect the number of visible
+  items in the list.
+- The input container should have \`role="combobox"\` and
+  \`aria-haspopup="listbox"\`. These attributes correctly identify the control
+  to Assistive Technology.
+- The input should have \`aria-autocomplete="list"\`.
+  This attribute correctly identifies the control to Assistive Technology.
+- The input should have \`autocomplete="off"\`, \`autocapitalize="off"\`,
+  and \`spellcheck="false"\`. These attributes deactivate default browser
+  behavior that can interfere with autocomplete behavior.
+- The list should have \`role="listbox"\`.
+- When the list is hidden, the input container should have
+  \`aria-expanded="false"\`.
+- When the list is visible, the input container should have
+  \`aria-expanded="true"\`.
+- When the list is visible and an item is highlighted (with the arrow keys),
+  the text input should have \`aria-activedescendant\` set to the \`id\`
+  of the active item.
+- When a list item is highlighted (with the arrow keys), it should have
+  \`aria-selected="true"\`.
+- Each selectable list item should have \`role="option"\`.
+
+
+#### Requires Additional Engineering:
+In order to keep the Spark Autocomplete flexible enough to use in a wide
+  variety of use cases, some of the interactivity of the component is left
+  up to implementing teams.
+
+  - Filtering the list
+    - Display a message if a filter returns 0 results
+    - Highlight the matching text in the search results
+    - Update the contents of the associated \`aria-live\` element
+  - Showing the list
+    - set \`aria-expanded\` to \`true\` on the input container
+  - Hiding the list
+    - remove \`aria-selected\` from the active list item
+    - remove \`aria-activedescendant\` from the input
+    - set \`aria-expanded\` to false on the input container
+  - Selecting an item in the list by clicking it or pressing the Enter key.
+    - The highlighted item is identified by the value of
+    \`aria-activedescendant\` on the input.
+  - \`data-sprk-autocomplete-no-select\` can be applied to any result item
+  that you wish to omit from the arrow key highlighting behavior. For example
+  a "No results found" message or an item that is hidden by a filter.
+    `,
+    docs: { iframeHeight: 400 },
+  },
+};
+
+export const defaultStory = () => {
+  useEffect(() => {
+    autocomplete();
+  }, []);
+
+  return `
+    <div data-sprk-autocomplete="container">
+      <div aria-live="polite" class="sprk-u-ScreenReaderText"></div>
```

Comment:
Hey @bobwise, when we use the screenreadertext class we have text inside the element. Is it missing here because it is something the user needs to add?

Reply:
The text is updated whenever the number of displayed results changes. aria-live elements should almost always be blank when the page loads and then populated later with JavaScript.